doc/nrf/config_and_build/bootloaders_and_dfu/bootloader.rst

```diff
@@ -186,3 +190,17 @@ For example, the partition layout for :file:`zephyr/samples/hello_world` using M
 
 You can also store secondary slot images in external flash memory when using MCUboot.
 See :ref:`ug_bootloader_external_flash` for more information.
+
+.. _ug_bootloader_flash_static_requirement:
+
+Static partition requirement for DFU
+------------------------------------
+
+As long as you are not using Device Firmware Updates (DFU), you can use the dynamic allocation of memory partitions for either immutable or upgradable bootloaders.
+
+However, if you want to perform DFU, you must :ref:`define a static partition map <ug_pm_static>`.
+This is important for different reasons, including:
+
+* If the dynamically generated partitions change between builds, the newly generated application images may no longer be compatible with the bootloader programmed to device.
+* Adding an upgradable bootloader to an application changes its :ref:`flash memory partition layout <ug_bootloader_flash_mcuboot>` to accommodate first and second image slots for both immutable and upgradable bootloaders.
```

Comment:
Maybe we could explain this in detail in the first point:
`If the dynamically generated partitions change between builds, the newly generated application images may no longer be compatible with the bootloader programmed to device.  In that case, the newly build application image may not be bootable by the bootloader programmed onto the device.`

Reply:
In that case we may no longer need to keep 2nd and 3rd bullet points (they refer to the same root cause of the issue).